libs/hiraya/src/components/Button/style.ts

```diff
@@ -9,27 +9,31 @@ interface ButtonProps {
   buttonSizes?: keyof typeof buttonSizeStyle;
   isFullWidth?: boolean;
   icon?: ReactNode;
+  focus?: boolean;
+  disabled?: boolean;
 }
 
 const ButtonStyled = styled(MUIButton)<ButtonProps>(
-  ({ variants, buttonSizes, isFullWidth, icon }) => ({
-    backgroundColor: variantStyle[variants || 'primary'].background,
-    color: variantStyle[variants || 'primary'].text,
-    padding: '0',
+  ({ variants, buttonSizes, isFullWidth, icon, focus, disabled }) => ({
+    ...buttonSizeStyle[buttonSizes || 'large'],
 
     '&.MuiButton-root': {
-      minWidth: icon
-        ? buttonSizeStyle[buttonSizes || 'large']
-        : isFullWidth
-        ? '112px'
-        : '92px',
-      height: buttonSizeStyle[buttonSizes || 'large'],
+      width: isFullWidth ? '100%' : buttonSizeStyle[buttonSizes || 'large'],
     },
 
+    ...(disabled
+      ? { ...disabledState }
+      : { ...variantStyle[variants || 'primary'] }),
```

Comment:
I dont exactly understand what you meant but this seems cleaner? also prevents unnecessary code when the button is disabled.